src/dune_rules/warning_emit.mli

```diff
@@ -0,0 +1,13 @@
+(** Emit warnings that respect dune's conventions *)
+
+open Import
+
+type context =
+  | Build of Path.Build.t
+  | Source of Source_tree.Dir.t
+  | Source_dir of Path.Source.t
```

Comment:
Why do we need both `Source` and `Source_dir` constructors? Can you rename one of them or add a comment explaining when to use one vs the other?

Reply:
Different parts of the code will find it easier to provide on value over another. I don't think I'm using all of these contexts though, so I will remove the ones that are unused.

Reply:
> Different parts of the code will find it easier to provide on value over another.

That makes sense. I think renaming the constructors to more closely match the types of their arguments will help to clarify their intent.